qiskit/pulse/schedule.py

```diff
@@ -438,6 +516,8 @@ def _add_timeslots(self, time: int, schedule: ScheduleComponent) -> None:
                         "".format(new=schedule.name or '', old=self.name or '', time=time,
                                   ch=channel, t0=interval[0], tf=interval[1]))
 
+        _check_nonnegative_timeslot(self._timeslots)
```

Comment:
This is inefficient. Why not just `if interval[0] < 0: / error` on line 509??

Reply:
This method is reused elsewhere. 509 is not the only place this check is relevant, also 496 and 248. Keeping track of all the places this could go wrong is much easier if its done in a separate pass. This is from experience trying to fix bugs with negative insertion time of a schedule.

Reply:
Definitely worth running some profiling, I could share my notebook with you so you have a couple values to compare against